src/tags/models.py

```diff
@@ -1,8 +1,27 @@
 from django.db import models
 
+from django.apps import apps
 from taggit.managers import TaggableManager
 
 
+def total_xp_by_tags(user, tags: list):
```

Comment:
Haven't thought about it yet, but more likely this will be a queryset of tags. I think your code below will be fine, but forcing a list here might be a problem.  Maybe just indicate in the docstring the expected type for now,

Reply:
added ``expects a list of tags for tags arg (try list(tags))`` to doc string